### PR TITLE
`Paywalls`: open Terms and Privacy Policy links in-app

### PR DESCRIPTION
**Pull request by aboedo**
Updated the terms and Privacy Policy links to open within the app, without exiting to Safari

https://github.com/RevenueCat/purchases-ios/assets/685609/e10df16d-1016-4271-9270-35c4ae8a87d4

### CODE REVIEW

**Comment by tonidero** [comment]
> what if we always open the links in a fullscreen modal?

I agree that makes more sense. Just not sure if you can present a modal on top of a modal, in case the developer is presenting the paywall as a modal already?

**Comment by aboedo** [comment]
Yeah, the modal thing might feel weird if you're already in a modal. One alternative is to make this remotely controllable from a variable

**Comment by NachoSoto** [comment]
How does this look on iPad, and visionOS?

**Review by NachoSoto — COMMENTED** [review]
I think this makes sense and it's a better UX 👍🏻

File: RevenueCatUI/Views/FooterView.swift [hunk 1]
```diff
@@ -257,6 +257,23 @@ private struct LinkButton: View {
 
 }
 
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(watchOS, unavailable)
```
[the remark this answers is not in the document]
Comment (NachoSoto) [reply RevenueCatUI/Views/FooterView.swift:261]: And maybe `visionOS`?

**Comment by NachoSoto** [comment]
Just watched the video. Perhaps we should present this as a sheet. Right now it shows a navigation bar which doesn't make a lot of sense since the controller doesn't have a title.

File: RevenueCatUI/Views/FooterView.swift [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
 import RevenueCat
 import SwiftUI
+import WebKit
```
Comment (NachoSoto) [inline RevenueCatUI/Views/FooterView.swift:16]: This doesn't exist on watchOS 😅

File: RevenueCatUI/Views/FooterView.swift [hunk 3]
```diff
@@ -221,12 +228,44 @@ private struct LinkButton: View {
...
+                                self.displayLink = false
+                            } label: {
+                                Image(systemName: "xmark")
+                            }
+                        }
+                    }
+            }
+        }
+        #else
+        Link(destination: self.url) {
```
Comment (NachoSoto) [inline RevenueCatUI/Views/FooterView.swift:255]: Keeping the existing implementation if `WebKit` isn't supported.

**Comment by NachoSoto** [comment]
Updated to use a sheet. Presenting a sheet on top of a sheet is a common pattern supported by iOS (see video)

**Review by NachoSoto — APPROVED** [review]

**Comment by stuart-j-williams** [comment]
This is broken on iPad as it opens the target in the sidebar of a split screen.  Do I need to log a new issue for this?  Thanks
IMG_0003 2

**Comment by NachoSoto** [comment]
Oh thanks! We'll get that fixed 🙏🏻

**Comment by NachoSoto** [comment]
@stuart-j-williams I'm unable to reproduce this, could you please file an issue including:
- iOS version
- Xcode version
- Code with how you're presenting the paywall.

Thanks!

**Pull request merged**